src/Form/Form.js

```diff
@@ -705,7 +726,8 @@ class Form {
 
         const mainType = getInputMainType(input)
         const subtype = getInputSubtype(input)
-        if (await this.device.settings.canAutofillType(mainType, subtype)) {
+        await this.device.settings.populateDataIfNeeded({ mainType, subtype })
+        if (this.device.settings.canAutofillType({ mainType, subtype }, this.device.inContextSignup)) {
```

Comment:
@GioSensation I've pulled the side affect out of `canAutofillType` so that it can be called synchronously, and existing functions have been updated to call the side effect (`populateDataIfNeeded`) beforehand. This should keep the existing functionality the same, while unlocking being able to call `canAutofillType` from the handler. Thoughts?

Reply:
That is AWESOME 🎉! I'm glad I've raised this and you've found a better solution. It was always weird to have an async function to check if we can autofill. One minor note is that I find the name `isUnfillable` confusing. Is it just non-native me? I'll reach out on MM.